### PR TITLE
Acme Challenge management

### PR DESCRIPTION
**Pull request by maraino**
### Description

This PR adds support for managing the ACME provisioner challenges using `step ca provisioner add` and `step ca provisioner update`.

### CODE REVIEW

File: command/ca/provisioner/add.go [hunk 1]
```diff
@@ -761,3 +776,49 @@ func createSCEPDetails(ctx *cli.Context) (*linkedca.ProvisionerDetails, error) {
...
+	case linkedca.Provisioner_SCEP:
+		if len(ctx.StringSlice("challenge")) > 1 {
+			return errors.New("provisioner type 'SCEP' does not support multiple '--challenge' flags")
+		}
+	}
+	return nil
+}
+
+// acmeChallengeToLinkedca returns the linkedca challenge types on the challenge
+// flag. It won't fail or add unsupported flags, the function assumes they
```
Comment (dopey) [inline command/ca/provisioner/add.go:807]: they -> the

File: command/ca/provisioner/provisioner.go [hunk 2]
```diff
@@ -314,21 +314,61 @@ disable ssh use '--ssh=false'.`,
...
+	// ACME and SCEP provisioner flags
 	forceCNFlag = cli.BoolFlag{
 		Name:  "force-cn",
 		Usage: `Always set the common name in provisioned certificates.`,
 	}
-	requireEABFlag = cli.BoolFlag{
+
+	challengeFlag = cli.StringSliceFlag{
+		Name: "challenge",
+		Usage: `With a SCEP provisioner is the <challenge> to use as a shared secret between a
```
Comment (dopey) [inline command/ca/provisioner/provisioner.go:325]: With a SCEP provisioner the <challenge> is a shared secret between the client and the CA.

**Review by dopey — APPROVED** [review]
lgtm. Let a few comments for documentation updates.

File: command/ca/provisioner/provisioner.go (hunk 2 above)
Comment (maraino) [reply command/ca/provisioner/provisioner.go:325]: Yes

Comment (maraino) [reply command/ca/provisioner/provisioner.go:325]: @dopey Should we rename `--remove-acme-challenge` to `--remove-challenge`, for consistency with the `--challenge` flag, even that doesn't make sense for scep?

Comment (dopey) [reply command/ca/provisioner/provisioner.go:325]: Yes, I think we should. Unless you want to add `add-acme-challenge`.

Comment (dopey) [reply command/ca/provisioner/provisioner.go:325]: Also, I meant that we should update the sentence in the line I linked. The grammar is weird.

Comment (maraino) [reply command/ca/provisioner/provisioner.go]: Fixed with f3bcd8a

File: command/ca/provisioner/add.go (hunk 1 above)
Comment (maraino) [reply command/ca/provisioner/add.go]: Fixed with f3bcd8a

**Comment by maraino** [comment]
@dopey Changed `--remove-acme-challenge` to `--remove-challenge` for consistency with `--challenge`, `--remove-challenge` is currently ACME only, it doesn't have any effect on SCEP.

**Review by dopey — APPROVED** [review]
lgtm

**Pull request merged**